### PR TITLE
Balancebot: Support for Acro, Guided, Auto and RTL modes

### PR DESCRIPTION
**Pull request by EbinPhilip**
- added a parameter for offsetting the balance point(equilibrium angle). This helps when the bot is not symmetric
- Minor balance bot specific code changes to support acro, auto, guided and rtl modes
- Added acro mode to MIS_DONE_BEHAVE

### CODE REVIEW

File: APMrover2/Parameters.cpp [hunk 1]
```diff
@@ -680,10 +680,19 @@ const AP_Param::GroupInfo ParametersG2::var_info[] = {
     // @Param: MIS_DONE_BEHAVE
     // @DisplayName: Mission done behave
     // @Description: Mode to become after mission done
-    // @Values: 0:Hold,1:Loiter
+    // @Values: 0:Hold,1:Loiter, 2:Acro
     // @User: Standard
     AP_GROUPINFO("MIS_DONE_BEHAVE", 38, ParametersG2, mis_done_behave, 0),
 
+    // @Param: BAL_ZERO_OFFSET
+    // @DisplayName: Balance zero offset
```
Comment (khancyr) [inline APMrover2/Parameters.cpp]: Add balance bot only !

File: APMrover2/Parameters.cpp [hunk 2]
```diff
@@ -680,10 +680,19 @@ const AP_Param::GroupInfo ParametersG2::var_info[] = {
...
+    // @Description: Zero point for balancing. Used when center of mass is not along roll axis. This offsets the tilt of the center of mass.
```
Comment (khancyr) [inline APMrover2/Parameters.cpp]: Add balance bot only !

File: libraries/AP_NavEKF3/AP_NavEKF3.h [hunk 3]
```diff
@@ -445,7 +445,7 @@ class NavEKF3 {
     const float gndEffectBaroScaler = 4.0f;        // scaler applied to the barometer observation variance when ground effect mode is active
     const uint8_t gndGradientSigma = 50;           // RMS terrain gradient percentage assumed by the terrain height estimation
     const uint16_t fusionTimeStep_ms = 10;         // The minimum time interval between covariance predictions and measurement fusions in msec
-    const uint8_t sensorIntervalMin_ms = 50;       // The minimum allowed time between measurements from any non-IMU sensor (msec)
+    const uint8_t sensorIntervalMin_ms = 20;       // The minimum allowed time between measurements from any non-IMU sensor (msec)
```
Comment (khancyr) [inline libraries/AP_NavEKF3/AP_NavEKF3.h]: This one will need evaluation in order to check that we don't speed to much time to  handle wheel encoder in depend of the rest

Comment (EbinPhilip) [reply libraries/AP_NavEKF3/AP_NavEKF3.h]: I had tried the original value of 50. There is a visible performance boost when reduced to 20. But I am not sure if there are some unseen consequences for other sensors. Nothing I could see though.

**Comment by EbinPhilip** [comment]
@khancyr I've made the changes in the comments

**Comment by khancyr** [comment]
thanks ! It should prevent some wrong usage ... I hope ...

**Comment by khancyr** [comment]
Seems OK to me. It doens't bother to have ACRO after Auto, even if I think that loiter is better.

**Comment by rmackay9** [comment]
Merged with some changes, thanks!

The changes I made were:

- renamed the new parameter to BAL_PITCH_TRIM
- stop_vehicle modified a bit to remove the call to get_throttle_out_stop for balance bots
- changed the prefix on the commits

thanks!

**Comment by rmackay9** [comment]
This was on the list to backport to Rover-3.5 but I'm afraid the merge is a bit challenging so I think we will need to leave it to Rover-3.6.

**Pull request closed without merging**